contracts/src/abstracts/AbstractPortal.sol

```diff
@@ -61,6 +61,20 @@ abstract contract AbstractPortal is IPortal {
     attestationRegistry.attest(attestationPayload, getAttester());
   }
 
+  /**
+   * @notice Attest the schema with given attestationPayload and validationPayload
+   * @param attestationPayload the payload to attest
+   * @param validationPayloads the payloads to validate via the modules to issue the attestations
+   * @dev Runs all modules for the portal and registers the attestation using AttestationRegistry
+   */
+  function attestV2(AttestationPayload memory attestationPayload, bytes[] memory validationPayloads) public payable {
+    moduleRegistry.runModulesV2(modules, attestationPayload, validationPayloads, msg.value, msg.sender, getAttester());
+
+    _onAttest(attestationPayload, getAttester(), msg.value);
```

Comment:
I would also suggest adding `validationPayloads` to the `_onAttest` function so that the user could do an `ECDSA` check or other from the portal itself. Allows for greater flexibility.